src/management-system-v2/components/modeler-share-modal-option-public-link.tsx

```diff
@@ -0,0 +1,207 @@
+import { Button, Checkbox, Col, Flex, Input, message, QRCode, Row, Typography } from 'antd';
+import { DownloadOutlined, CopyOutlined, LoadingOutlined } from '@ant-design/icons';
+import { SetStateAction, useEffect, useState } from 'react';
+import { useParams } from 'next/navigation';
+
+const ModelerShareModalOptionPublicLink = () => {
+  const { processId } = useParams();
+  const [token, setToken] = useState(null);
+  const [isShareLinkChecked, setIsShareLinkChecked] = useState(false);
+  const [onlyAsRegisteredUser, setOnlyAsRegisteredUser] = useState(false);
+  const [publicLinkValue, setPublicLinkValue] = useState(
+    `${window.location.origin}/shared-viewer?token=`,
+  );
+
+  const handleCopyLink = async () => {
+    try {
+      await navigator.clipboard.writeText(publicLinkValue);
+      message.success('Link Copied');
+    } catch (err) {
+      message.error('Error copying link');
+    }
+  };
+
+  const generateToken = async () => {
+    try {
+      const response = await fetch('/api/share/generate-share-token', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          registeredUsersOnly: onlyAsRegisteredUser,
+          processId: processId,
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to generate token');
+      }
+
+      const { token } = await response.json();
+      message.success('Token generated successfully');
+      setToken(token);
+      setPublicLinkValue(`${window.location.origin}/shared-viewer?token=${token}`);
+    } catch (error) {
+      message.error('Error generating token');
+    }
+  };
+
+  useEffect(() => {
```

Comment:
When possible avoid useEffect ([see here](https://react.dev/learn/you-might-not-need-an-effect)). In this case the token generation can be moved to handleShareLinkChecked